Library/Homebrew/cmd/readall.rb

```diff
@@ -69,4 +74,17 @@ def readall
       end
     end
   end
+
+  def syntax_errors_or_warnings?(rb)
+    # Syntax errors are printed to stderr and result in a non-zero exit code.
+    # Syntax warnings are also printed to stderr, but don't alter the exit code.
+    if Open3.respond_to?(:capture3)
+      _out, err, status = Open3.capture3(RUBY_PATH, "-c", "-w", rb)
+      $stderr.print err
+      !status.success? || !err.chomp.empty?
+    else
+      # Fallback for Ruby 1.8.7. Doesn't flag syntax warnings as failure.
+      system RUBY_PATH, "-c", "-w", rb
```

Comment:
We need to capture `stderr` (normally discarded by `popen_read`) instead of `stdout`. I guess we could make that

```ruby
Utils.popen_read("ruby -c -w #{rb} 2>&1 > /dev/null")
```

to keep it compatible, if you feel strongly about this. I have a personal distaste for solutions where I'm forced to leave argument splitting to some intermediate shell (think whitespace in file names). That's why I implemented it like this. (And in the hopes that dropping 1.8 is a not too distant future.)